src/pipelinedefinition.cpp

```diff
@@ -223,91 +223,214 @@ void PipelineDefinition::makeSubscriptions(ModelManager& manager) {
     }
 }
 
-Status PipelineDefinition::validateNode(ModelManager& manager, NodeInfo& node) {
-    std::unique_ptr<ModelInstanceUnloadGuard> nodeModelInstanceUnloadGuard;
-    std::shared_ptr<ModelInstance> nodeModelInstance;
-    tensor_map_t nodeInputs;
-    SPDLOG_DEBUG("Validation of pipeline: {} node: {} type: {}", getName(), node.nodeName, node.kind);
-
-    Status result;
-    if (node.kind == NodeKind::DL) {
-        result = getModelInstance(manager, node.modelName, node.modelVersion.value_or(0), nodeModelInstance,
-            nodeModelInstanceUnloadGuard);
-        if (!result.ok()) {
-            SPDLOG_ERROR("Validation of pipeline({}) definition failed. Missing model: {} version: {}", this->pipelineName, node.modelName, node.modelVersion.value_or(0));
-            return StatusCode::MODEL_NAME_MISSING;
+Status PipelineDefinition::validateNode(ModelManager& manager, const NodeInfo& dependantNodeInfo) {
+    std::unique_ptr<ModelInstanceUnloadGuard> dependantModelUnloadGuard;
+    std::shared_ptr<ModelInstance> dependantModelInstance;
+    std::set<std::string> remainingUnconnectedDependantModelInputs;
+    SPDLOG_DEBUG("Validation of pipeline: {}; node name: {}; node kind: {}",
+        getName(),
+        dependantNodeInfo.nodeName,
+        dependantNodeInfo.kind);
+
+    // If we validate DL model node, retrieve underlying model instance.
+    if (dependantNodeInfo.kind == NodeKind::DL) {
```

Comment:
Maybe separate method for validation of specific thigs for NodeKind::DL?

This method is already long and hard to comprehend without deep focus.
I am looking for ways how we could encapsulate how much we can.

Reply:
This would create bunch of new methods and all would need to have plenty of variables passed over like: unload guard, underlying model instance, current nodeinfo, source node info and connection map. I've tried to comment each step, maybe it is enough?

Reply:
Separated validation into NodeValidator class. Separated parts of validation code into smaller methods.